src/librustc/lint/mod.rs

```diff
@@ -710,11 +718,16 @@ pub fn struct_lint_level<'a>(sess: &'a Session,
                 );
                 sess.diag_note_once(&mut err, diag_msg_id, &msg);
             }
+
+            pre_warn_level
         }
-    }
+    };
 
-    if raised {
-        let msg = format!("raised to `{}`, the minimum level for `{}`", level.as_str(), name);
+    // Highlight the minimum as cause of the lint iff it was raised due to the minimum.
+    due_to_raised_by_minimum |= pre_warn_level.filter(|&pwl| lint.min_level > pwl).is_some();
+    if due_to_raised_by_minimum {
+        let level_str = lint.min_level.as_str();
+        let msg = format!("#[{}({})] is the minimum lint level", level_str, name);
```

Comment:
The previous form of this message, "raised to $level, the minimum level for $lint", made it pretty clear that the minimum itself is derived *from* the lint itself.

In my opinion, the new form of the message is less clear on that front; it is just stating a fact, with less of a hint about that fact's derivation. 

Thus, a user may sit there trying to figure out whether this is something they can adjust via some directive in the source code or a compiler flag.

Reply:
I debated this a lot... I agree that the "raised ..." formulation is good but imo it made little sense together with the `#[deny(illegal_floating_point_literal_pattern)] implied by #[deny(warnings)]` form which is why I changed it.

How about adding `the lint level cannot be reduced to Allow` as is done for the `unused_attributes` warning?

Reply:
> How about adding `the lint level cannot be reduced to $level` as is done for the `unused_attributes` warning?

okay, sounds good